lib/wda-mac.js

```diff
@@ -282,21 +295,74 @@ class WDAMacServer {
     }
   }
 
+
+  /**
+   * @typedef {Object} ProxyProperties
+   *
+   * @property {string} scheme - The scheme proxy to.
+   * @property {string} host - The host name proxy to.
+   * @property {number} port - The port number proxy to.
+   * @property {string} path - The path proxy to.
+   */
+
+  /**
+   * Returns proxy information where WDAMacServer proxy to.
+   *
+   * @param {Object} caps - The capabilities in the session.
+   * @return {ProxyProperties}
+   */
+  parseProxyProperties (caps) {
+    let scheme = 'http';
+    if (!caps.webDriverAgentMacUrl) {
+      return {
+        scheme,
+        server: (this.process?.host ?? caps.systemHost) ?? DEFAULT_SYSTEM_HOST,
+        port: (this.process?.port ?? caps.systemPort) ?? DEFAULT_SYSTEM_PORT,
+        path: ''
+      };
+    }
+
+    const {protocol, hostname, port, pathname} = url.parse(caps.webDriverAgentMacUrl);
+    if (_.isString(protocol)) {
+      scheme = protocol.split(':')[0];
+    }
+    return {
+      scheme,
+      server: hostname ?? DEFAULT_SYSTEM_HOST,
```

Comment:
I would say we should throw an exception if hostname cannot be parsed

Reply:
same about protocol

Reply:
Will do this later

Reply:
I did this raising an exception in https://github.com/appium/appium-mac2-driver/pull/41/files#diff-cb3604d19b5cec159d0260eb8b8739d21bf074cf54584fd2b555e9969caaa92fR335-R341 (url.URL)